guides/common/modules/proc_renewing-a-custom-ssl-certificate-on-server.adoc

```diff
@@ -0,0 +1,55 @@
+[id="Renewing_a_Custom_SSL_Certificate_on_{project-context}_{context}"]
+= Renewing a {customssltitle} certificate on {ProjectServer}
+
+[NOTE]
+====
+If {ProjectServer} is already configured with CA-signed SSL certificates, and it is renewed by the same Certificate Authority, avoid replacing the consumer package `katello-ca-consumer-latest.noarch.rpm` on clients registered to {ProjectServer}.
+
+If the validity of CA-signed SSL certificates installed on {Project} from Certificate Authority A expires, and you installed an SSL certificate from Certificate Authority B, you must re-deploy the `katello-ca-consumer-latest.noarch.rpm` package on every client registered to {Project}.
+You need a new CA certificate for a client to authenticate or communicate with {ProjectServer}.
+====
+
+Use this procedure to update your renewed {customssl} certificate signed by a Certificate Authority (CA) for {ProjectServer}.
+The `katello-certs-check` command validates the input certificate files and returns the commands necessary to renew a {customssl} certificate for {ProjectServer}.
+
+.Prerequisite
+Refer to the {InstallingServerDocURL}Configuring_Server_with_a_Custom_SSL_Certificate_{context}[Configuring {ProjectServer} with a Custom SSL Certificate] guide before creating a new CSR because the Server certificate must have X.509 v3 `Key Usage` and `Extended Key Usage` extensions with required values.
+
+.Procedure
+You must create a new Certificate Signing Request (CSR) and send it to the Certificate Authority to sign the certificate.
+In return, you will receive the {ProjectServer} certificate and CA bundle.
+
+Before deploying a renewed custom certificate on your {ProjectServer}, follow these steps:
+
+* Validate the {customssl} input files.
+Note that for the `katello-certs-check` command to work correctly, Common Name (CN) in the certificate must match the FQDN of {ProjectServer}:
++
+[options="nowrap" subs="+quotes,attributes"]
+----
+# katello-certs-check -t {project-context} \
+-b /root/_{project-context}_cert/ca_cert_bundle.pem_ \
+-c /root/_{project-context}_cert/{project-context}_cert.pem_ \
+-k /root/_{project-context}_cert/{project-context}_cert_key.pem_
+----
```

Comment:
I am inclined to leave this in because users see running the installer as a "major" action that can result in service downtime. I believe they get annoyed when they get ready to take that downtime and it fails, and now they have to either get a new outage window or extend the current one while they debug their certificate issue.

Reply:
Because of https://projects.theforeman.org/issues/36567 we need to leave it in for now. Right now we can't rely on the installer halting. In the future we can drop it.

As for the downtime argument: I didn't consider that. IMHO it's something to address. Users should not be scared to run the installer.

Reply:
> As for the downtime argument: I didn't consider that. IMHO it's something to address. Users should not be scared to run the installer.

I think is a good, bigger conversation.

Reply:
We could consider an enhancement (not blocking to this change) to add a "--check-certs" type option to the installer that would align the interface for users to be the installer but allow them to check the certs without executing the rest of the installer.